app/scripts/services/dimManifestService.factory.js

```diff
@@ -0,0 +1,149 @@
+(function() {
+  'use strict';
+
+  angular.module('dimApp')
+    .factory('dimManifestService', ManifestService);
+
+  ManifestService.$inject = ['$rootScope', '$q', 'dimBungieService', '$http'];
+
+  function ManifestService($rootScope, $q, dimBungieService, $http) {
+    // Testing flags
+    const alwaysLoadRemote = false;
+
+    let manifestPromise = null;
+
+    return {
+      getManifest: function() {
+        if (manifestPromise) {
+          return manifestPromise;
+        }
+
+        // TODO: var language = window.navigator.language;
+        var language = 'en';
+
+        manifestPromise = dimBungieService.getManifest()
+          .then(function(data) {
+            var version = data.version;
+
+            return loadManifestFromCache(version)
+              .catch(function(e) {
+                var path = data.mobileWorldContentPaths[language] || data.mobileWorldContentPaths.en;
+                return loadManifestRemote(version, path);
+              })
+              .then(function(typedArray) {
+                return new SQL.Database(typedArray);
+              });
+          });
+
+        return manifestPromise;
+      },
+
+      getRecord: _.memoize(function(db, table, id) {
+        // The ID in sqlite is a signed 32-bit int, while the id we use is unsigned, so we must convert
+        const sqlId = new Int32Array([id])[0];
+        const result = db.exec(`SELECT json FROM ${table} where id = ${sqlId}`);
+        if (result.length && result[0].values && result[0].values.length) {
+          return JSON.parse(result[0].values[0]);
+        }
+        return null;
+      }, (db, table, id) => table + '-' + id),
+
+      getAllRecords: function(db, table) {
+        const rows = db.exec(`SELECT json FROM ${table}`);
+        const result = {};
+        rows[0].values.forEach((row) => {
+          const obj = JSON.parse(row);
+          result[obj.hash] = obj;
+        });
+        return result;
+      }
+    };
+
+    /**
+     * Returns a promise for the manifest data as a Uint8Array. Will cache it on succcess.
+     */
+    function loadManifestRemote(version, path) {
+      return $http.get("https://www.bungie.net/" + path, { responseType: "blob" })
+        .then(function(response) {
+          return unzipManifest(response.data);
+        })
+        .then(function(arraybuffer) {
+          var typedArray = new Uint8Array(arraybuffer);
+          var encoded = base64js.fromByteArray(typedArray);
+          return $q(function(resolve, reject) {
+            // Note: this requires the 'unlimitedStorage' permission, as the
```

Comment:
I'm not sure - I assume the permissions popup says something like "Store unlimited data"?